### PR TITLE
reaction actions

### PR DESCRIPTION
**Pull request by matthewdias**
I don't know how this affects activities.

_"Every action has an equal and opposite reaction..."_

### CODE REVIEW

File: app/templates/media-reactions.hbs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div class="col-md-8 col-xs-11 centered">
+  {{media/media-reaction reaction=model}}
```
Comment (trmcnvn) [inline app/templates/media-reactions.hbs]: Shouldn't we use the feed component for the permalink page?

File: app/components/media/media-reaction.js [hunk 2]
```diff
@@ -4,14 +4,23 @@ import set from 'ember-metal/set';
...
   isUpvoted: false,
-  isEditable: false,
+  showUser: true,
   queryCache: service(),
   store: service(),
+  router: service('-routing'),
   tasksRunning: or('getUserVoteTask.isRunning', 'createVoteTask.isRunning', 'destroyVoteTask.isRunning'),
 
+  canModerate: getter(function() {
+    if (get(this, 'session.account').hasRole('admin', get(this, 'reaction'))) {
```
Comment (trmcnvn) [inline app/components/media/media-reaction.js]: We should check if the user is the owner here too.

File: app/components/media/media-reaction.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+export default Component.extend(ClipboardMixin, {
+  isUpvoted: false,
+  showUser: true,
+  queryCache: service(),
+  store: service(),
+  router: service('-routing'),
+  tasksRunning: or('getUserVoteTask.isRunning', 'createVoteTask.isRunning', 'destroyVoteTask.isRunning'),
+  host: getter(() => `${location.protocol}//${location.host}`),
+
+  canDelete: getter(function() {
```
Comment (trmcnvn) [inline app/components/media/media-reaction.js]: getter -> computed.

File: app/components/media/media-reaction.js [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+import ClipboardMixin from 'client/mixins/clipboard';
+
+export default Component.extend(ClipboardMixin, {
+  isUpvoted: false,
+  showUser: true,
+  queryCache: service(),
+  store: service(),
+  router: service('-routing'),
+  tasksRunning: or('getUserVoteTask.isRunning', 'createVoteTask.isRunning', 'destroyVoteTask.isRunning'),
+  host: getter(() => `${location.protocol}//${location.host}`),
```
Comment (trmcnvn) [inline app/components/media/media-reaction.js]: Don't want to continue the use of `getter`, we can just set this in `init`

**Review by trmcnvn — CHANGES_REQUESTED** [review]

**Comment by trmcnvn** [comment]
So many conflicts :scream:

**Comment by matthewdias** [comment]
oh my god what happened

**Review by trmcnvn — APPROVED** [review]

**Comment by trmcnvn** [comment]
More conflicts :P

**Pull request merged**